dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/DateUnitPeriodTypeParser.java

```diff
@@ -117,7 +117,7 @@ private DateInterval parseInternal(
               week,
               calendar,
               PeriodType.MAP_WEEK_TYPE.get(periodType.getName()),
-              new DateTimeUnit(year, 1, 1));
+              periodType.adjustToStartOfWeek(new DateTimeUnit(year, 1, 4), calendar));
```

Comment:
Unclear: Why are we adjusting, why use `4` as parameter?

Reply:
In ISO week, the first week of a year should contain Jan 4.

Reply:
Sorry for late response, I wasn't very clear with my comment. I didn't want an explanation of the line of code, I wanted the code to be updated in a way so it becomes clear reading it what we are doing and why. In the simplest form this could be a comment on the branch or line. In this case it should be clear why you want to check the 4. of January of all days. There must be a reason.